### ISSUE TITLE
Are the options for `parmopts` documented anywhere?

### ISSUE THREAD

**Issue by slochower** [Amber-MD/cpptraj#1057]
Are the options for `parmopts` explicitly listed anywhere (specifically for `strip`)? I'm looking for a way to prevent `strip` from renumbering residues, so that subsequent analyses on the stripped topology retains the original numbering. I searched the manual for `parmopts` and don't see what's available. Is it the same options that the regular `parm` command accepts?

**Comment by drroe** [comment]
Hi,

> Are the options for parmopts explicitly listed anywhere (specifically for strip)?

Sorry, it's not very clear; `<parmopts>` in this case refers to options for writing the topology, so what you would get from `help Formats parmwrite` in the interactive help or what's listed in the `parmwrite` section of the Cpptraj manual.

>  I'm looking for a way to prevent strip from renumbering residues, so that subsequent analyses on the stripped topology retains the original numbering.

The `:` operator will always use internal indexing, so that will always reflect the current sequence. Try the `:;` operator instead (the so-called "PDB" numbering) since that should remain unchanged. Type `help Mask` into the interactive or check the manual for more details. Hope this helps!

**Comment by slochower** [comment]
> The : operator will always use internal indexing, so that will always reflect the current sequence. Try the :; operator instead (the so-called "PDB" numbering) since that should remain unchanged. Type help Mask into the interactive or check the manual for more details. Hope this helps!

My understanding is this works within the same "session" but once you save the stripped `prmtop/nc` to further processing, the original numbering will be lost (because it's not encoded in the `prmtop`, right)? So, say I have a multistep workflow where the first step extracts the protein-ligand system into stripped `prmtop/nc`s. Then later, I read in those files and run `atomicfluct` but want the results to reference the original numbering for easy interpretability. That's not possible, right? (An other way to handle this, I think, would be to save a mapping between the original and "stripped" `prmtop` after running `strip`.)

**Comment by rodrigogalindo** [comment]
David, perhaps you can use the change command (
https://amberhub.chpc.utah.edu/change/) to re-number your stripped files.

**Comment by slochower** [comment]
@rodrigogalindo That could work. What would be the syntax for renumbering back to the original? I don' tunderstand why `oresnums` takes a `min` and `max`. Could you use `min :1` and `max :99999` to just do everything?

**Comment by rodrigogalindo** [comment]
I think if you use '*'  it will change everything. The reason for min
and max is for you to start, for example :1-317 min 32 max 348 will
change 1 to 32 and so on. In your case, just use the 'min' as the
start number of your lowest residue ID, Best way is to try out :)

**Comment by slochower** [comment]
Is there a way to do this without knowing the min and max residue numbers, because that may change system to system with waters, ions, etc. For example, if I try `change parmindex 0 oresnums of *` then I see 
```
  [change parmindex 0 oresnums of *]
        Using topology: xxx.cms-in.pdb
        * selects 20133 residues.
        Original (output) res #s: 0 to 0 (1)
Error: # selected residues (20133) != # provided residue numbers (1).
        1 errors encountered reading input.
```

**Comment by slochower** [comment]
Actually, looking back on things, I can see why that doesn't work -- that option will change the original residue numbers. But at the same time, I can't see a way to change the existing residue numbers (back to the original ones or to something else). So I'm not sure that using `change` is a way to renumber the stripped file.

**Comment by drroe** [comment]
Hi David, 

> My understanding is this works within the same "session" but once you save the stripped prmtop/nc to further processing, the original numbering will be lost

This is no longer the case as of version 4.29.0 or so. Cpptraj will use/add the `RESIDUE_NUMBER` flag to an Amber topology file, which tracks the "original" residue number. You can see this works by doing a brief test in the `$CPPTRAJHOME/test` directory. First do:
```
parm tz2.parm7
trajin tz2.rst7
distance d1 :5 :6 out temp.d1.dat
strip :1-4 parmout temp.tz2.strip.parm7
trajout temp.tz2.strip.rst7
run
```
then run
```
parm temp.tz2.strip.parm7
trajin temp.tz2.strip.rst7
distance d1 :;5 :;6 out temp.d2.dat
run
```
The values in temp.d1.dat and temp.d2.dat are identical since residues 5 and 6 do not change. Does this make sense?

**Comment by slochower** [comment]
Yes! That makes sense. I see how you can use `:;` to select via the original residue numbers, but can you make tools output using the original residue numbers, too (e.g., `atomicfluct`)?

**Comment by slochower** [comment]
Yes, perfect. This is working perfectly now. Would you like me to make a PR for the documentation?

**Comment by slochower** [comment]
#1059

**Comment by drroe** [comment]
Addressed by #1059, thanks @slochower !
